modules/system/common/hardware/powersave/power-tuning.sh

```
@@ -0,0 +1,325 @@
+cpu_governor=$(cat /sys/devices/system/cpu/cpu0/cpufreq/scaling_governor 2>/dev/null || echo "unknown")
+EOF
+
+    # Backup PCIe ASPM if available
+    if [[ -f /sys/module/pcie_aspm/parameters/policy ]]; then
+        echo "pcie_aspm=$(cat /sys/module/pcie_aspm/parameters/policy)" >> "$SETTINGS_FILE"
+    fi
+
+    # Backup runtime PM settings for some common devices
+    for device in /sys/bus/pci/devices/*/power/control; do
+        if [[ -f "$device" ]]; then
+            local device_path=$(dirname "$device")
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)